src/main/java/seedu/duke/Storage.java

```diff
@@ -1,10 +1,113 @@
 package seedu.duke;
 
+import seedu.duke.model.Patient;
+import seedu.duke.model.Record;
+
+import java.io.FileWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
 /**
  * This class handles the loading and saving of data from/to a plaintext file on
  * the hard drive.
- * TODO: To be implemented
  */
 public class Storage {
-    
+    protected String filePath;
+
+    /**
+     * This is the constructor of the Storage class.
+     * @param filePath A String of the path to the file that is used for saving/loading
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Saves a SortedMap of data into the file specified by filePath.
+     * @param patientData The data to be written to file
+     * @throws IOException When file is not found etc
+     */
+    public void save(SortedMap<String, Patient> patientData) throws IOException {
+        try {
+            File inFile = new File(filePath);
+            inFile.createNewFile();
+            FileWriter fileWriter = new FileWriter(inFile.getAbsolutePath(), false);
+            StringBuffer message = new StringBuffer();
```

Comment:
I think everything from this part until `fileWriter.close()` doesn't need to be in the try catch block, could you double-check that?

Reply:
Since the fileWriter uses inFile, there will be an error if I put it outside the try catch block. I'll try to see if there's anything I can do about this. Thanks!

Reply:
No worries, I just checked and I think that writing to file could also throw an `IOException`, so it's fine to leave it in the try-catch.